backend/routes/officers.py

```diff
@@ -51,19 +55,35 @@ def search_officer():
     logger = logging.getLogger("officers")
 
     try:
-        if body.officerName:
+        if body.name:
             names = body.officerName.split()
-            first_name = names[0] if len(names) > 0 else ''
-            last_name = names[1] if len(names) > 1 else ''
-            query = Officer.query.filter(or_(
-                Officer.first_name.ilike(f"%{first_name}%"),
-                Officer.last_name.ilike(f"%{last_name}%")
-            ))
+            if len(names) == 0:
```

Comment:
This would be an empty string. We would never accept this case, would we? I think this case and next should have their values incremented. On the `len(names) == 1` case, we would experience an out of bounds exception when indexing `names[1]`.